lib/notifications/db/notification.ex

```diff
@@ -68,4 +68,120 @@ defmodule Notifications.Db.Notification do
     )
     |> validate_required(:created_at)
   end
+
+  defmodule Queries do
+    @moduledoc """
+    Composable queries for accessing `Notifications.Db.Notification`
+    related data
+    """
+    import Ecto.Query
+
+    @doc """
+    The "base" query that queries `Notifications.Db.Notification`'s without restriction
+    """
+    def base() do
+      from(n in Notifications.Db.Notification, as: :notification, select_merge: n)
+    end
+
+    def select_user_read_state(query \\ base(), user_id \\ nil) do
+      from([notification: n] in query,
+        join: nu in assoc(n, :notification_users),
+        as: :notification_user,
+        join: u in assoc(nu, :user),
+        as: :user,
+        where: u.id == ^user_id,
+        select_merge: %{
+          state: nu.state
+        }
+      )
+    end
+
+    @doc """
+    Joins associated `Notifications.Db.Detour`'s on
+    `Notifications.Db.Notification`'s and retrieves the Detour's
+    associated info.
+
+    ## Example
+
+    The `query` parameter defaults to
+    `Notifications.Db.Notification.Queries.base()`, but can be specified
+    explicitly
+
+        iex> Notifications.Db.Notification.Queries.base()
```

Comment:
This looks like a doctest, but I don't think it's running, and am not sure it would pass. 

Is it worth it to set up the plumbing required to make this a runnable doctest?

Reply:
We could pair on this? I don't think the contents of the return value is important, so I'd benefit from some pairing/guidance there, but I do like the idea of running doctests here